### PR TITLE
sort by version rather than just time stamp

### PR DESCRIPTION
**Pull request by johnoliver**
##### Checklist

- [x] `mvn clean package` build and test completes
- [ ] documentation is changed or added

### CODE REVIEW

File: adoptopenjdk-api-v3-frontend/src/test/kotlin/net/adoptopenjdk/api/AssetsResourceFeatureReleasePathTest.kt [hunk 1]
```diff
@@ -2,20 +2,13 @@ package net.adoptopenjdk.api
...
 import io.restassured.RestAssured
-import junit.framework.Assert.assertTrue
+import junit.framework.Assert.fail
 import net.adoptopenjdk.api.v3.JsonMapper
 import net.adoptopenjdk.api.v3.dataSources.SortOrder
-import net.adoptopenjdk.api.v3.models.Architecture
-import net.adoptopenjdk.api.v3.models.ImageType
-import net.adoptopenjdk.api.v3.models.OperatingSystem
-import net.adoptopenjdk.api.v3.models.ReleaseType
+import net.adoptopenjdk.api.v3.models.*
```
Comment (karianna) [inline adoptopenjdk-api-v3-frontend/src/test/kotlin/net/adoptopenjdk/api/AssetsResourceFeatureReleasePathTest.kt]: Prefer explicit imports

File: adoptopenjdk-api-v3-frontend/src/test/kotlin/net/adoptopenjdk/api/AssetsResourceFeatureReleasePathTest.kt [hunk 2]
```diff
@@ -63,43 +56,40 @@ class AssetsResourceFeatureReleasePathTest : AssetsPathTest() {
...
-        val times = getTimestamps(SortOrder.ASC)
-
-        times
-                .fold(LocalDateTime.MIN, { previous, updatedAt ->
-                    assertTrue("${previous} is after ${updatedAt}", updatedAt.isAfter(previous) || updatedAt.equals(previous))
-                    updatedAt
+        getReleases(SortOrder.ASC)
+                .fold(null, { previous: Release?, next: Release ->
+                    if (previous != null) {
+                        if (previous.version_data.compareTo(next.version_data) > 0) {
```
Comment (karianna) [inline adoptopenjdk-api-v3-frontend/src/test/kotlin/net/adoptopenjdk/api/AssetsResourceFeatureReleasePathTest.kt]: What happens with =?

File: adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/VersionData.kt [hunk 3]
```diff
@@ -92,4 +92,17 @@ class VersionData {
         return true
     }
 
+    override fun compareTo(other: VersionData): Int {
+        return COMPARITOR.compare(this, other)
+    }
+
+    companion object {
+        val COMPARITOR = compareBy<VersionData> { it.major }
```
Comment (karianna) [inline adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/VersionData.kt]: `Comparator`

File: adoptopenjdk-api-v3-frontend/src/test/kotlin/net/adoptopenjdk/api/AssetsResourceFeatureReleasePathTest.kt (hunk 2 above)
Comment (johnoliver) [reply adoptopenjdk-api-v3-frontend/src/test/kotlin/net/adoptopenjdk/api/AssetsResourceFeatureReleasePathTest.kt]: if we ever release 2 releases with the same version, and include it in the test set, then this test fails

**Pull request merged**